### PR TITLE
implement cards back-end

### PR DESCRIPTION
**Pull request by vladimirDobrik**

### CODE REVIEW

**Comment by vladimirDobrik** [comment]
[implement back-end for cards](https://trello.com/c/88J9KgVj/57-implement-back-end-for-cards)

File: server/src/controller/card/card.controller.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            return response.status(500).json(error);
+        }
+    }
+
+    @httpPost('/save-card/')
+    public async saveCard(request: Request, response: Response){
+        const card = request.body;
+
+        try {
+            const isSave = await this.cardService.saveCard(card.id, card.name, card.surname, card.image, card.skills, card.hp, card.damage);
```
Comment (vharadkou) [inline server/src/controller/card/card.controller.ts]: please remove isSave const, you already have try/catch

File: server/src/service/card/card.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { inject, injectable } from 'inversify';
+
+import { Card } from '../../models/new-card';
+import { CardRepository } from './card.repository';
+import { LoggerService } from '../logger';
+
+@injectable()
+export class CardService {
```
Comment (vharadkou) [inline server/src/service/card/card.service.ts]: For what purpose you created this service, it is just unnecessary layer with try/catch and logging.
You already have logging on repository level, and try/catch in controller
Please remove this service

File: server/src/controller/card/card.controller.ts [hunk 3]
```diff
@@ -24,11 +23,10 @@ export class CardController {
...
 
         try {
-            const isSave = await this.cardService.saveCard(card.id, card.name, card.surname, card.image, card.skills, card.hp, card.damage);            
-            if (isSave) {                
-                response.status(200).send({status: 'Saved'});
-            } else {
-                response.status(400).send({status: 'Error'});
+            await this.cardRepository.saveCard(card.id, card.name, card.surname, card.image, card.skills, card.hp, card.damage);
+
+            if (response.statusCode == 200) {
```
Comment (vharadkou) [inline server/src/controller/card/card.controller.ts]: ??? you never set response status code, please remove unnecessary "if", just return response with corresponding status

File: server/src/controller/card/card.controller.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import { CardRepository } from './../../service/card/card.repository';
+
+@controller('/api')
+export class CardController {
+    public constructor(
+        @inject(CardRepository) private cardRepository: CardRepository
+    ) { }
+
+    @httpGet('/get-cards/')
+    public async getSkills(request: Request, response: Response){
```
Comment (vharadkou) [inline server/src/controller/card/card.controller.ts]: return type

File: server/src/controller/card/card.controller.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    public async getSkills(request: Request, response: Response){ 
+        try {
+            return await this.cardRepository.getCards();
+          } catch (error) {
+            return response.json(error);
+        }
+    }
+
+    @httpPost('/save-card/')
+    public async saveCard(request: Request, response: Response){
```
Comment (vharadkou) [inline server/src/controller/card/card.controller.ts]: return type

File: server/src/controller/card/card.controller.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+@controller('/api')
+export class CardController {
+    public constructor(
+        @inject(CardRepository) private cardRepository: CardRepository
+    ) { }
+
+    @httpGet('/get-cards/')
+    public async getSkills(request: Request, response: Response){ 
+        try {
+            return await this.cardRepository.getCards();
```
Comment (vharadkou) [inline server/src/controller/card/card.controller.ts]: you can remove await, in async function all return promises will automatically waitable

**Pull request merged**